_posts/en/newsletters/2020-06-24-newsletter.md

```diff
@@ -147,8 +147,11 @@ BOLTs][bolts repo].*
 
 - [Bitcoin Core #19260][] p2p: disconnect peers that send filterclear + update existing filter msg disconnect logic FIXME:dongcarl
 
-- [Bitcoin Core #19133][] rpc, cli, test: add bitcoin-cli -generate command FIXME:moneyball
-
+- [Bitcoin Core #19133][] restores the bitcoin-cli `generate` command
+  which had been removed from Bitcoin Core. It is a shortcut for
+  `generatetoaddress(getnewaddress())` and is useful for testing with
+  regtest.
```

Comment:
`generate` used to be a server-side RPC and (according to https://github.com/bitcoin/bitcoin/issues/14299) was removed for software architectural reasons.

One reason for its return is many of the historical tutorials were broken when it was removed (per https://github.com/bitcoin/bitcoin/pull/17700#issuecomment-566159096).

It was brought back as a client-side CLI `-generate` command (denoted with a preceding dash), outside of the RPC API because that is considered the right place for CLI commands for use by humans, for example per https://github.com/bitcoin/bitcoin/pull/16439#issuecomment-516926539.

I have to write a release note for it now; will share the link here.

Reply:
Release note https://github.com/bitcoin/bitcoin/pull/19357/files